bson/bsoncodec/default_value_decoders.go

```diff
@@ -111,7 +111,7 @@ func (dvd DefaultValueDecoders) BooleanDecodeValue(dctx DecodeContext, vr bsonrw
 
 	var b bool
 	var err error
-	switch vr.Type() {
+	switch vrType := vr.Type(); vrType {
```

Comment:
It was done to avoid multiple calls to `vr.Type()` and because I prefer this style over calling the same method repeatedly. Changed in all decoders for consistency.